### PR TITLE
Add flag and warning for when hdbscan not installed

### LINKED ISSUE #197: HDBScan numpy C header error
Reproducing the error.
```
pip install numpy==1.19.2 hdbscan
python -c "import hdbscan"
```
throws
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/nicksorros/code/test/pip_problem/venv/lib/python3.7/site-packages/hdbscan/__init__.py", line 1, in <module>
    from .hdbscan_ import HDBSCAN, hdbscan
  File "/Users/nicksorros/code/test/pip_problem/venv/lib/python3.7/site-packages/hdbscan/hdbscan_.py", line 21, in <module>
    from ._hdbscan_linkage import (single_linkage,
  File "hdbscan/_hdbscan_linkage.pyx", line 1, in init hdbscan._hdbscan_linkage
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
```

This is issue is mentioned in hdbscan issue tracker https://github.com/scikit-learn-contrib/hdbscan/issues/457

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

Fixes #197 by delegating to the user to install hdbscan if they wish to use it and warning them about it.

It would be good to add tests both for the flag and for hbscan. Possible need to update documentation of README as well.

We could install them for them as we do for sent2vec in the case that they did not do themselves and tried to use it.

Checklist
---

- [x] Added link to Github issue or Trello card
- [ ] Added tests

### CODE REVIEW

**Comment by nsorros** [comment]
[Resolve WellcomeML errors in master and branches](https://trello.com/c/3GIiOwCl/1170-resolve-wellcomeml-errors-in-master-and-branches)

File: wellcomeml/ml/clustering.py [hunk 1]
```diff
@@ -2,19 +2,27 @@
...
 from sklearn.cluster import DBSCAN, OPTICS, KMeans
 from sklearn.manifold import TSNE
 from sklearn.model_selection import GridSearchCV
 from sklearn.metrics import silhouette_score
 from sklearn.pipeline import Pipeline
-#  from hdbscan import HDBSCAN
+try:
+    from hdbscan import HDBSCAN
+    HDBSCAN_INSTALLED = True
+except ValueError:
```
Comment (aCampello) [inline wellcomeml/ml/clustering.py:18]: I'm really surprised this is not a `ImportError`

File: wellcomeml/ml/clustering.py [hunk 2]
```diff
@@ -2,19 +2,27 @@
...
+    HDBSCAN_INSTALLED = False
+    logger.warning(
+        "If you want to use hdbscan you need to run"
+        "pip3 install hdbscan --no-cache-dir --no-binary :all: --no-build-isolation"
```
Comment (aCampello) [inline wellcomeml/ml/clustering.py:22]: Maybe a link to the Issue here.

**Comment by aCampello** [comment]
I like the idea, I think it is on the right track, but the build doesn't passed as yet...

File: wellcomeml/ml/clustering.py (hunk 1 above)
Comment (nsorros) [reply wellcomeml/ml/clustering.py:18]: You can try it out by commenting out the flags in the install
`$(VIRTUALENV)/bin/pip3 install hdbscan #--no-cache-dir --no-binary :all: --no-build-isolation`

Any way we can add a test for this?

**Review by aCampello — APPROVED** [review]

**Pull request merged**